core/domain/exp_domain.py

```diff
@@ -554,6 +565,15 @@ def validate(self, obj_type, exp_param_specs_dict):
 
         self.outcome.validate()
 
+    def get_fuzzy_rule_spec(self):
```

Comment:
Maybe we should just return the index here, instead of the rule_spec. Then the caller can lookup the rule_spec using the returned index. It seems cleaner to me. WDYT?